client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -201,14 +237,45 @@ function CommentableEditor({commentApp, fieldNode, contentPath, rawContentState,
 
     return <DraftailEditor
     ref={editorRef}
-    onChange={setEditorState}
+    onChange={(editorState) => {
+      let newEditorState = editorState;
+      if (['undo', 'redo'].includes(editorState.getLastChangeType())) {
+        const filteredContent = filterInlineStyles(inlineStyles.map(style => style.type).concat(ids.map(id => COMMENT_STYLE_IDENTIFIER + id)), editorState.getCurrentContent());
+        newEditorState = forceResetEditorState(editorState, filteredContent);
+      }
+      setEditorState(newEditorState);
+    }
+    }
     editorState={editorState}
     controls={enabled ? [CommentControl] : []}
     decorators={enabled ? [{
-      strategy: findCommentStyleRanges,
+      strategy: (block, callback) => findCommentStyleRanges(block, callback),
```

Comment:
Is this really needed?

Reply:
This is because otherwise it gets a third argument (which isn't the `filterFn` argument) - `ContentState` I believe